core/blockchain.go

```diff
@@ -1376,6 +1392,9 @@ func (bc *BlockChain) writeBlockWithState(block *types.Block, receipts []*types.
 	rawdb.WriteBlock(blockBatch, block)
 	rawdb.WriteReceipts(blockBatch, block.Hash(), block.NumberU64(), receipts)
 	rawdb.WritePreimages(blockBatch, state.Preimages())
+	if (randomCommitment != common.Hash{}) {
+		rawdb.WriteRandomCommitmentCache(blockBatch, randomCommitment, block.ParentHash())
```

Comment:
If my understanding is correct, a validator must process the last block they proposed to include the randomness in the DB?

Reply:
Yes, they will process it to find out the resulting state.  TBH, I'm not really understanding your concern about the block processing and this code.

Reply:
This was me working out that we still need the backward iteration, but just at startup time, and that this handles the normal case operation of saving the randomness when processing blocks.